src/client/tile.cpp

```diff
@@ -1812,6 +1812,157 @@ bool TextureSource::generateImagePart(std::string part_of_name,
 			baseimg->drop();
 			baseimg = img;
 		}
+		/*
+			[item:modname:itemname:WxH
+			Renders an item to texture, width and height are optional
+		*/
+		else if (part_of_name.substr(0,6) == "[item:") {
+			if (baseimg != NULL) {
+				errorstream << "generateImagePart(): baseimg != NULL "
+						<< "for part_of_name=\"" << part_of_name
+						<< "\", cancelling." << std::endl;
+				return false;
+			}
+			if (!driver->queryFeature(video::EVDF_RENDER_TO_TARGET)) {
+				errorstream << "generateImagePart(): No rendertarget support "
+						<< "for part_of_name=\"" << part_of_name
+						<< "\", cancelling." << std::endl;
+				return false;
+			}
+
+			assert(m_itemdef_manager && m_client);
+
+			Strfnd sf(part_of_name);
+			sf.next(":");
+			std::string modname = sf.next(":");
+			std::string itemname = sf.next(":");
+			const std::string item = modname + ":" + itemname;
+			const std::string w = sf.next("x");
+			u32 width = g_settings->getU16("item_rtt_speedup_resolution");
+			u32 height = width;
+			if (w != "") {
+				width = stoi(w);
+				const std::string h = sf.next("");
+				if (h != "")
+					height = stoi(h);
+				else
+					height = width;
+			}
+
+			const ItemDefinition &def = m_itemdef_manager->get(item);
+			ItemMesh *imesh = m_itemdef_manager->getWieldMesh(item, m_client);
+
+			if (!imesh || !imesh->mesh) {
+				errorstream << "generateImagePart(): No mesh available for item "
+						<< "for part_of_name=\"" << part_of_name
+						<< "\", cancelling." << std::endl;
+				return false;
+			}
+
+			const core::dimension2d<u32> img_dim(width, height);
+			const core::rect<s32> res_rect(0, 0, width, height);
+
+			core::rect<s32> oldViewPort = driver->getViewPort();
+			core::matrix4 oldProjMat = driver->getTransform(video::ETS_PROJECTION);
+			core::matrix4 oldViewMat = driver->getTransform(video::ETS_VIEW);
+			core::matrix4 oldWorldMat = driver->getTransform(video::ETS_WORLD);
+
+			video::ITexture* rt = driver->addRenderTargetTexture(img_dim, "render_invimg", video::ECF_A8R8G8B8);
+			if (!rt) {
+				errorstream << "generateImagePart(): Generating rendertarget failed "
+						<< "for part_of_name=\"" << part_of_name
+						<< "\", cancelling." << std::endl;
+				return false;
+			}
+			driver->setRenderTarget(rt, true, true, video::SColor(0,0,0,0));
+
+			scene::IMesh *mesh = imesh->mesh;
+			driver->clearZBuffer();
+
+			core::matrix4 ProjMatrix;
+			ProjMatrix.buildProjectionMatrixOrthoLH(2.0f, 2.0f, -1.0f, 100.0f);
+
+			core::matrix4 ViewMatrix;
+			ViewMatrix.buildProjectionMatrixOrthoLH(2.0f, 2.0f, -1.0f, 100.0f);
+			ViewMatrix.setTranslation(core::vector3df(
```

Comment:
Why copy & paste? `tile.cpp` and `hud.cpp` are both client files, thus it would make sense to move this code to a separate function which both files can access.

Reply:
@pyrollo noted the same.

> code redundancy still needs to be fixed